src/isomorphic/hooks/ReactComponentTreeHook.js

```diff
@@ -49,57 +64,52 @@ var canUseMap = (
 );
 
 if (canUseMap) {
```

Comment:
Instead of having each function handle both cases, I made a different function for each one. I find it to be easier to read and understand what's going on, and I believe it should be a tiny bit faster :)